guardian/commercial-templates
# SPIKE: Migrate interscroller template to Svelte

## What does this change?
Migrated the interscroller template to Svelte

The `post` function in `messenger.ts` needed some tweaks to actually be able to work in production. Frontend discards messages that are not strings and ones that do not provide an an id in the format of a UUID.

## How to test
The interscroller makes use of extensive code in frontend (basically all the interscrolling happens there) so unfortunately would be quite difficult to preview using `yarn dev`.

We have setup a test line item accessed with `?adtest=interscroller_test_220727` which you can use on a mobile viewport in a `top-above-nav` slot (first inline).

It should function identically to the legacy template accessable under a different adtest `?adtest=DAP_samung_inter_pg`.

You can verify you're looking at the svelte version by looking for the comment that's inserted at the top:
```
/* "interscroller" updated on 2022-07-27 via https://github.com/guardian/commercial-templates/blob/87a777443/src/templates/ssr/interscroller/index.svelte */
```
